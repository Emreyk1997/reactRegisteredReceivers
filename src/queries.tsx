// import gql from 'graphql-tag';
// import { useQuery } from '@apollo/react-hooks';

// export const query = gql`
//   query Books {
//     books {
//       title
//       author
//     }
//   }
// `;

// export default () => useQuery(query);

import gql from "graphql-tag";
import { useQuery } from '@apollo/react-hooks';

export const GET_COUNTER = gql`
  query GetCounterValue {
    counter @client
  }
`;

export default () => useQuery(GET_COUNTER);
