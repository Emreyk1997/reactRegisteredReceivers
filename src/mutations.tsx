import gql from "graphql-tag";
import { GET_COUNTER } from "./queries";
import {logger} from './logger';

export const UPDATE_COUNTER = gql`
mutation updateCounter($offset: Number!) {
  updateCounter(offset: $offset) @client
}`;

export const CounterMutations = {
  updateCounter: (_, variables, { cache }) => {
    //query existing data
    const data = cache.readQuery({ query: GET_COUNTER });
    //Calculate new counter value
    const newCounterValue = data.counter + variables.offset;
    console.log('NEWCOUNTER');
    logger.info(variables);
    logger.info(newCounterValue);
    cache.writeData({
      data: { counter: newCounterValue }
    });
    return null; //best practice
  }
};