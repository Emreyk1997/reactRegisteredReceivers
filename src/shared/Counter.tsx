import React, { useState } from 'react';
import { logger } from '../logger';
import { useQuery, useMutation } from '@apollo/react-hooks';
import { GET_COUNTER, GET_CONFIG } from '../state/queries/counterQueries';
import { UPDATE_COUNTER } from '../state/mutations/counterMutation';
// import { useInjection } from '../providers/ioc.react';
// import { IProvider } from '../providers/providers';

const Counter: React.SFC = () => {
  const {data: dataR, loading: loadingR} = useQuery(GET_CONFIG);
  const { data, loading, error } = useQuery(GET_COUNTER);
  const [, setErrorCatch] = React.useState(null);
  const [increment] = useMutation(UPDATE_COUNTER, { variables: { offset: 1 } });
  const [decrement] = useMutation(UPDATE_COUNTER, { variables: { offset: -1 } });
  // const provider = useInjection<IProvider<string>>('nameProvider');

  const logIncrement = () => {
    logger.error('ERROR');
    logger.info('INFO');
    logger.warn('WARN');
    logger.debug('DEBUG');
    console.log('INCREMENTTTTT');
    increment();
  };

  if (loading) return <p>Loading...</p>;

  if (error) return <p>Error</p>;
  if (data?.counter < 5) {
    console.log('DATAAS', data, dataR)
    return (
      <div>
        <p>Provider:</p>
        {/* <p>{provider.provide()}</p> */}
        <p data-testid="counter" >Counter: {data?.counter}</p>
        <p>Process: {dataR.process}</p>
        <button data-testid="button-up" type="button" onClick={() => logIncrement()}>
          Increase
        </button>
        <button data-testid="button-down" type="button" onClick={() => decrement()}>
          Decrease
        </button>
      </div>
    );
  } else {
    logger.error('Error from Boundary');
    // setErrorCatch(() => {
    //   throw new Error('error');
    // });
  }
};

export default Counter;
