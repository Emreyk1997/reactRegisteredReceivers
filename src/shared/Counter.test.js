import React from 'react';
import { render, cleanup, fireEvent, waitForElement } from '@testing-library/react';
import Counter from './Counter';
import { MockedProvider } from '@apollo/react-testing';
import { GET_COUNTER, GET_CONFIG } from '../state/queries/counterQueries';
import { UPDATE_COUNTER } from '../state/mutations/counterMutation';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom/extend-expect';

// import * as Injection from '../providers/ioc.react';
// import * as IProvider from '../providers/providers';

// jest.mock('../providers/ioc.react');
// jest.mock('../providers/providers');
let deleteMutationCalled = false;

const CounterMocks = [
  {
    request: {
      query: GET_COUNTER,
    },
    result: {
      data: {
        counter: 0,
      },
    },
  },
  {
    request: {
      query: UPDATE_COUNTER,
      variables: { offset: 1 },
    },
    result: () => {
      deleteMutationCalled = true;
      return {
        data: {
          counter: 1,
        },
      };
    },
    // error: new Error('aw shucks'),
  },
  {
    request: {
      query: GET_CONFIG,
    },
    result: {
      data: {
        process: 'hello',
      },
    },
  },
];
// const ConfigMocks = [
//   {
//     request: {
//       query: GET_CONFIGS,
//     },
//     result: {
//       data: {
//         process: 'hello',
//       },
//     },
//   },
// ];

describe('Take a snapshot', () => {
  afterEach(cleanup);

  it('should take a snapshot', async () => {
    const promise = Promise.resolve();
    const { asFragment } = render(
      <MockedProvider mocks={CounterMocks} addTypename={false}>
        <Counter />
      </MockedProvider>
    );

    expect(
      asFragment(
        <MockedProvider mocks={CounterMocks} addTypename={false}>
          <Counter />
        </MockedProvider>
      )
    ).toMatchSnapshot();
    await act(() => promise);
  });

  it('increments counter', async () => {
    const promise = Promise.resolve();
    const { getByTestId } = render(
      <MockedProvider mocks={CounterMocks} addTypename={false}>
        <Counter />
      </MockedProvider>
    );
    const greetingData = await waitForElement(() => getByTestId('button-up'));
    // const button = getByTestId('button-up');
    fireEvent.click(getByTestId('button-up'));
    // await wait(3000);
    // button.props.onClick();
    const greetingDataCounter = await waitForElement(() => getByTestId('counter'));
    expect(deleteMutationCalled).toBe(true);
    // expect(greetingDataCounter).toHaveTextContent('1');
    await act(() => promise);
  });
});
