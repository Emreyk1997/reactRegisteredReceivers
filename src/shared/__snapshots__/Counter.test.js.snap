// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Take a snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;
